### PR TITLE
Add instructions about how to write Argo workflows for E2E tests.

### PR DESCRIPTION
**Pull request by jlewi**
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/testing/258)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -543,3 +517,233 @@ Webhooks for prow should already be configured according to these [instructions]
...
+
+       * **app_dir**: Is the path to the ksonnet directory within the repository. This should be of the form `${GITHUB_ORG}/${GITHUB_REPO_NAME}/${PATH_WITHIN_REPO_TO_KS_APP}`
+       * **component**: This is the name of the ksonnet component to use for the Argo workflow
+       * **name**: This is the base name to use for the submitted Argo workflow. A unique suffix is added to give each run a unique name.
+       * **job_types**: This is an array specifying for which types of [prow jobs](https://github.com/kubernetes/test-infra/blob/master/prow/jobs.md)
+          this workflow should be triggered on.
+
+          * Currently allowed values are **presubmit**, **postsubmit**, and **periodic**.
+
+      * **include_dirs**: If specified, the pre and postsubmit jobs will only trigger this test if the PR changed at least one file matching at least one
```
Comment (richardsliu) [inline README.md]: We should call out that periodic tests ignore this flag.

File: README.md [hunk 2]
```diff
@@ -543,3 +517,233 @@ Webhooks for prow should already be configured according to these [instructions]
...
+            - bar/*
+              params:
+          params:
+            platform: gke
+            gkeApiVersion: v1beta1
+       ```
+
+       * **app_dir**: Is the path to the ksonnet directory within the repository. This should be of the form `${GITHUB_ORG}/${GITHUB_REPO_NAME}/${PATH_WITHIN_REPO_TO_KS_APP}`
+       * **component**: This is the name of the ksonnet component to use for the Argo workflow
+       * **name**: This is the base name to use for the submitted Argo workflow. A unique suffix is added to give each run a unique name.
```
Comment (richardsliu) [inline README.md]: I think there are some limitations to the length of the name, might be worth mentioning here.

File: README.md (hunk 1 above)
Comment (jlewi) [reply README.md]: Done.

File: README.md (hunk 2 above)
Comment (jlewi) [reply README.md]: Done. I suggest checking the length in the ksonnet file and adjusting it as needed. In the section on the Argo spec I explicitly call out the 63 character limit on Argo workflow names.

**Comment by jlewi** [comment]
@richardsliu PTAL.

**Pull request merged**